src/script/src/python/pyo3/vector_impl.rs

```diff
@@ -299,6 +299,51 @@ impl PyVector {
             })?;
             let ret = Self::from(ret).into_py(py);
             Ok(ret)
+        } else if let Ok(slice) = needle.downcast::<PySlice>(py) {
+            let indices = slice.indices(self.len() as i64)?;
+            let (start, stop, step, slicelength) = (
+                indices.start,
+                indices.stop,
+                indices.step,
+                indices.slicelength,
+            );
+            if start < 0 {
+                return Err(PyValueError::new_err(format!(
+                    "Negative start is not supported, found {start} in {indices:?}"
+                )));
+            } // Negative stop is supported, means from "indices.start" to the actual start of the vector
+            let vector = self.as_vector_ref();
+
+            let mut buf = vector
+                .data_type()
+                .create_mutable_vector(indices.slicelength as usize);
+            let v = if indices.slicelength == 0 {
+                buf.to_vector()
+            } else if step == 1 {
+                vector.slice(start as usize, slicelength as usize)
+            } else {
+                if indices.step > 0 {
+                    let range = if stop == -1 {
```

Comment:
can this handle `v[:-2]`?

BTW, for simplicy, I think we need not special handle `1` (like step = 1 or stop = -1), and leave it to rust `Range`. We only need to take care of negative ranges. WDYT

Reply:
> can this handle `v[:-2]`?
> 
> BTW, for simplicy, I think we need not special handle `1` (like step = 1 or stop = -1), and leave it to rust `Range`. We only need to take care of negative ranges. WDYT

After call `slice.indices(len)` stop is already adjusted to positive indices( So stop==-1 actually means before array's head), but it's reasonable to add related tests, gonna add some